s30900-Pawel_Kozlowski/Zadania/14.10.2023 Procesy i Strumienie Zadania.txt

```diff
@@ -0,0 +1,387 @@
+Zadanie 0
+
+1.Przejdź do swojego katalogu domowego
+   "cd ~" (umiem)
+
+2.Wydaj komendę: ls -a (umiem)
+   Po wpisaniu komendy otrzymałem: (wypisało wszystko wraz z ukrytymi plikami)
+
+   osboxes  ~  ls -a
+   .              .gksu.lock                 Templates
+   ..             .gnome                     test
+   .asoundrc      .gnupg                     .thumbnails
+   .bash_history  .gtk-bookmarks             ukos
+   .bash_logout   .gtkrc-2.0                 .vboxclient-clipboard.pid
+   .bashrc        .ICEauthority              .vboxclient-display-svga-x11.pid
+   .cache         .lesshst                   .vboxclient-draganddrop.pid
+   .config        .local                     .vboxclient-seamless.pid
+   .dbus          .mozilla                   Videos
+   Desktop        Music                      .Xauthority
+   .dmrc          Pictures                   .xsession-errors
+   Documents      .profile                   .xsession-errors.old
+   Downloads      Public
+   .face          .sudo_as_admin_successful
+
+3.Program wypisał 40 plików (rozumiem)
+
+4.Teraz wydaj komendę: ls -a | grep D
+   (Po wpisaniu polecenia, otrzymałem pliki zaczynające się na literę D)
+   osboxes ~  ls -a | grep D
+   Desktop
+   Documents
+   Downloads
+
+5.Zobacz ile teraz jest wyników. Co się stało?(umiem)
+   ls -a - wypisało wszytkie pliki i katalogi i grep D wypisał katalogi i pliki które pasują do wzorca
+   odpowiedź 3 wyniki
+
+6.A teraz wykonaj komendę: ls -a | grep D > lista_plikow_z_literka_d.txt
+   osboxes  ~  ls -a | grep D > lista_plikow_z_literka_d.txt
+   wykonałem polecenie domyślam się że został utworzony nowy plik
+
+7.Zobacz czy utworzył sie jakiś plik?(proste)
+   w katalogu domowym został utworzony plik lista_plikow_z_literka_d.txt
+   osboxes  ~  ls
+   Desktop    Downloads                     Music     Public     test  Videos
+   Documents  lista_plikow_z_literka_d.txt  Pictures  Templates  ukos
+
+   Jaka jest jego treść?
+   w pliku znajdują się pliki które zaczynały się na literę D 
+   osboxes  ~  cat lista_plikow_z_literka_d.txt
+   Desktop
+   Documents
+   Downloads
+
+   Co znaczy | oraz co znaczy > ?
+   "|" - oznacza wykonanie jednego a zaraz po nim kolejnego polecenia(łączy polecenia)
```

Comment:
To błędny opis. To o czym piszesz to de facto operator `&&`.
Tu jest dodatkowo zrealizowane przekierowanie strumienia stdout na stdin następnego polecenia i to jest sednem tego operatora a nie to, że jedno polecenie wykona się za poprzednim